docs/inheritance_and_composition.md

```diff
@@ -0,0 +1,36 @@
+# Template Inheritance and Composition
+
+## What is Inheritance, what is Composition?
+
+Template Inheritance is a way to create a new template based on an existing one. It
+allows you to reuse code and structure from the parent template while adding or
+overriding specific parts in the child template. This is useful for creating variations
+of a template without duplicating code.
+
+<!-- TODO: add an example of inheritance -->
+
+Template Composition, on the other hand, is a way to combine multiple templates,
+independent of each other, into a single new template. In this way you can join the
+functionalities of multiple templates, by adding some "glue code", and avoid repetition.
+
+<!-- TODO: add an example of composition -->
+
+<!-- TODO? (not sure if it relevant...) add an operational distinction between the two -->
+
+## Why copier does not natively support inheritance and composition?
+
+<!-- TODO: add content, re-use content from https://github.com/copier-org/copier/issues/934#issuecomment-1518964035 -->
```

Comment:
I anticipate that a proper solution might require Copier to support it natively. At least for the fork-like workflow, I think that specified template versions in parent template migrations need to be mapped to child template versions – at the very least. But it's too early to make a definitive statement.

Reply:
Reading carefully what he wrote once again, it seems to me that everything reduces to the UNIX philosophy of "do one thing well", or am I wrong?